Tasks/ServiceFabricDeploy/Create-DiffPackage.ps1

```diff
@@ -0,0 +1,122 @@
+
+		# If $clusterServiceManifestNames is null, it means no services are running in the cluster. Diff Package is equal to Full Package. Use Full Package to do deployment
+		if (!$clusterServiceManifestNames)
+		{
+			Write-Host (Get-VstsLocString -Key DIFFPKG_NoServicesRunning -ArgumentList @($ApplicationName, $ConnectedServiceEndpoint.Url))
+			Return
+		}
+
+		Write-Host (Get-VstsLocString -Key DIFFPKG_CopyingToDiffPackge -ArgumentList @($localAppManifestPath, (Join-Path $diffPackagePath $appManifestName)))
+		Copy-Item $localAppManifestPath (Join-Path $diffPackagePath $appManifestName) -Force
+
+		# Get the service manifests from the cluster
+		$clusterServiceManifestByName = @{}
+		foreach ($clusterServiceManifestName in $clusterServiceManifestNames)
+		{
+			$clusterServiceManifestContent = Get-ServiceFabricServiceManifest -ApplicationTypeName $applicationTypeName -ApplicationTypeVersion $clusterAppTypeVersion -ServiceManifestName $clusterServiceManifestName
+			$clusterServiceManifestByName[$clusterServiceManifestName] = [XML]$clusterServiceManifestContent
+		}
+
+		foreach ($serviceManifestImport in $localAppManifestXml.ApplicationManifest.ServiceManifestImport)
+		{
+			# Open the service manifest associated with the current ServiceManifestImport element of the local ApplicationManifest
+			$serviceManifestName = "ServiceManifest.xml"
+			$localServicePkgPath = Join-Path $applicationPackagePath $serviceManifestImport.ServiceManifestRef.ServiceManifestName
+			$localServiceManifestPath = [System.IO.Path]::Combine($localServicePkgPath, $serviceManifestName)
+			$localServiceManifest = ([XML](Get-Content $localServiceManifestPath)).ServiceManifest
+			$diffServicePkgPath = [System.IO.Path]::Combine($diffPackagePath, $localServiceManifest.Name)
+			$clusterServiceManifest = $clusterServiceManifestByName[$localServiceManifest.Name].ServiceManifest
+
+			# If there's no matching manifest from the cluster it means this is a newly added service that doesn't exist yet on the cluster. 
+			if (!$clusterServiceManifest)
+			{
+				# Copy this service and all the children
+				Write-Host (Get-VstsLocString -Key DIFFPKG_ServiceDoesNotExist -ArgumentList @($localServiceManifest.Name, $ApplicationName, $ConnectedServiceEndpoint.Url))
+				Copy-Item $localServicePkgPath $diffServicePkgPath -Recurse
+				continue
+			}
+		
+			# If the Version of the Service is not upgraded, skip upgrading the Service
+			if ($clusterServiceManifest.Version -eq $localServiceManifest.Version)
+			{
+				Write-Host (Get-VstsLocString -Key DIFFPKG_ServiceIsNotChanged -ArgumentList @($localServiceManifest.Name, $ApplicationName, $clusterServiceManifest.Version, $ConnectedServiceEndpoint.Url))
+				continue
+			}
+
+			Copy-DiffPackage -clusterPackages $clusterServiceManifest.CodePackage -localPackages $localServiceManifest.CodePackage -localParentPkgPath $localServicePkgPath -diffParentPkgPath $diffServicePkgPath
```

Comment:
Copy-DiffPackage supports an array of packages. Either do this as one call for all three packages or remove the support for doing more than one package in Copy-DiffPackage.

Reply:
@NCarlsonMSFT Using array as the parameter handles the cases when there are multiple CodePackage/ConfigPackage. One sort of package is an array